doc/htmldoc/nest_sonata/nest_sonata_guide.rst

```diff
@@ -0,0 +1,259 @@
+organized by utilizing HDF5 groups and datasets. The SONATA format requires certain HDF5 datasets, for instance 
+a dataset containing all the node type ids. Properties that are 
+stored on an individual-basis, for instance synaptic weights, are also stored in HDF5 datasets. 
+
+Simulation parameters and the locations of the HDF5 and CSV files specifying the network are stored in JSON 
+configuration files. 
+
+
+.. _sec:sonata_nodes:
+
+The NEST support of SONATA nodes 
+--------------------------------
+
+In the SONATA format, node populations are serialized in nodes HDF5 files and have a single associated node types 
+CSV file that assigns properties to all nodes with a given node type id. A node types CSV file may be shared by 
+multiple node population HDF5 files.
+
+NEST assumes the following structure of the nodes HDF5 files: 
+
+:: 
+
+    <nodes_file.h5>                     Filename
+    ├─ nodes                            Group - required
+    │  ├─ <population_name>             Group - required - usually only one but can be more population groups per file
+    │  │  ├─ node_type_id               Dataset {N_total_nodes} - required
+
+**Note:** NEST assumes that the implicit row numbers in the ``node_type_id`` dataset correspond to the ``node_id``\s. 
+
+NEST supports the following SONATA node ``model_type``\s:
+
+* ``point_neuron``
+* ``point_process``
+* ``virtual`` 
+
+Both ``point_neuron`` and ``point_process`` mean that the node is a neuron model (explicitly simulated) whereas ``virtual`` 
+means that the node only provide inputs to the simulated system. ``virtual`` nodes are modelled as ``spike_generator``\s 
+(see :doc:`../models/spike_generator`\). NEST requires that only one ``model_type`` is present per node types CSV file. 
+
+The required headers for node types CSV files that describe neuron models are: 
+
+* ``node_type_id``
+* ``model_type``
+* ``model_template``
+* ``dynamics_params``
+
+For a given ``node_type_id``, the ``model_template`` entry is the name of the NEST neuron model with prefix ``nest:``. NEST 
+does not require the ``model_template`` entries to be the same, but the creation of the nodes described in a single node 
+types CSV file is faster if the neuron models are the same. 
+
+For a given ``node_type_id``, the ``dynamics_params`` entry is expected to be a reference to a JSON file that describes 
+the parametrization of the neuron model. Below is an example of a JSON file describing the parametrization of a given 
+node type: 
+
+.. code-block:: json
+
+    {
+        "I_e": 0.0,
+        "tau_m": 44.9,
+        "C_m": 239.0,
+        "t_ref": 3.0,
+        "E_L": -78.0,
+        "V_th": -43.0,
+        "V_reset": -55.0
+    }
+
+
+NEST does not support node properties stored on an individual-basis in HDF5 datasets. This restriction can be 
+circumvented by assigning a single node its own node type id. 
+
+Below is an example of a node types CSV file with the required headers for neuron nodes: 
```

Comment:
```suggestion
Below is an example of a node type CSV file with the required headers for neuron nodes: 
```

Reply:
All of  these plural 'types' read awkwardly to me.  It seems like it should be singular descriptor.

Reply:
I tend to agree. I based the spelling on the SONATA documentation which use a plural descriptor, but see now that the SONATA paper use a singular.